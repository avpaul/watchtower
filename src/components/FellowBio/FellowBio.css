.fellow-bio-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-left: 2em;
}

.card-bio {
  display: inline-block;
  height: 240px;
  width: 580px !important;
  border-radius: 4px;
  background-color: #ffffff;
  padding-top: 0px;
  width: 648px;
}
.card-bio:focus{
  outline: 1px solid #0459e4 !important;
}
.card-bio:after {
  content: '';
  display: table;
  clear: both;
}

.card-project {
  display: flex;
  display: inline-block;
  height: 240px;
  width: 580px !important;
  border-radius: 4px;
  background-color: #ffffff;
}
.card-project:focus{
  outline: 1px solid #0459e4 !important;
}
.rectangle {
  height: 30px;
  width: 4px;
  background-color: #ffaf30;
  margin: 27px 15px 30px 40px;
}

.my-bio {
  display: flex;
  height: 26px;
  width: 112px;
  color: #0459e4;
  font-family: 'DIN Pro';
  font-size: 20px;
  font-weight: bolder;
  line-height: 26px;
  margin-bottom: 25px;
}

.bio {
  display: flex;
  flex-direction: column;
  padding: 2em;
  width: 576px;
}

.my-project {
  height: 26px;
  width: 112px;
  color: #0459e4;
  font-family: 'DIN Pro';
  font-size: 20px;
  font-weight: bolder;
  line-height: 26px;
  min-width: 200px;
  margin-bottom: 25px;
}

.project {
  display: flex;
  flex-direction: column;
  width: 683px;
  padding: 2em;
}

.title {
  height: 26px;
  width: 461.38px;
  color: #232f49;
  font-family: 'DIN Pro';
  font-size: 20px;
  line-height: 26px;
  padding: 0px 0px 20px 30px;
  margin: 16px;
}

.content {
  position: absolute;
  height: 26px;
  width: 461.38px;
  color: #232f49;
  font-family: 'DIN Pro';
  font-size: 20px;
  line-height: 26px;
  margin: 29px 99.62px 182px 65px !important;
  word-spacing: 4px;
}

.spacing {
  margin: 0px 6px 0px 0px;
}
