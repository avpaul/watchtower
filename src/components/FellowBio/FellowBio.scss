@import '../../assets/sass/abstract/variables';

.fellow-bio-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.card-bio {
  display: inline-block;
  height: 240px;
  width: 550px !important;
  border-radius: 4px;
  background-color: #ffffff;
  padding-top: 0px;
  width: 648px;
  position: relative;
}

.card-bio:focus {
  outline: 1px solid #0459e4 !important;
}
.card-bio:after {
  content: '';
  display: table;
  clear: both;
}

.card-bio .loader__center,
.card-project .loader__center {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.card-project {
  display: flex;
  display: inline-block;
  height: 240px;
  width: 100% !important;
  border-radius: 4px;
  background-color: #ffffff;
  position: relative;
}
.card-project:focus {
  outline: 1px solid #0459e4 !important;
}
.rectangle {
  height: 30px;
  width: 4px;
  background-color: #ffaf30;
  margin: 27px 15px 30px 40px;
}

.my-bio {
  display: flex;
  height: 26px;
  width: 112px;
  color: #0459e4;
  font-family: $font-primary;
  font-size: 20px;
  font-weight: bolder;
  line-height: 26px;
  margin-bottom: 25px;
}

.bio {
  display: flex;
  flex-direction: column;
  padding: 0 2em 2em 0;
  width: 576px;
}

.my-project {
  height: 26px;
  width: 112px;
  color: #0459e4;
  font-family: $font-primary;
  font-size: 20px;
  font-weight: bolder;
  line-height: 26px;
  min-width: 200px;
  margin-bottom: 25px;
}

.project {
  display: flex;
  flex-direction: column;
  width: 550px;
  padding: 0 0 2em 0;
}

.title {
  height: 26px;
  width: 461.38px;
  color: #232f49;
  font-family: $font-primary;
  font-size: 20px;
  line-height: 26px;
  padding: 0px 0px 20px;
  margin: 16px;
}

.content {
  position: absolute;
  height: 26px;
  width: 461.38px;
  color: #232f49;
  font-family: $font-primary;
  font-size: 20px;
  line-height: 26px;
  margin: 29px 99.62px 182px 65px !important;
  word-spacing: 4px;
}

.spacing {
  margin: 0px 6px 0px 0px;
}
