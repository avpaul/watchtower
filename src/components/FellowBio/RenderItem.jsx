import React from 'react';
import PropTypes from 'prop-types';

const RenderItem = ({ title, children }) => (
  <div className="row ml-0 title">
    <div className="rectangle" />
    <div className="content">
      <b className="spacing">{title}:</b>
      {children}
    </div>
  </div>
);

RenderItem.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

RenderItem.defaultProps = {
  children: ''
};

export default RenderItem;
