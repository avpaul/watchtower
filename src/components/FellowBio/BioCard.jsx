import React from 'react';
import PropTypes from 'prop-types';
import RenderItem from './RenderItem';
import Loader from '../Loader/Loader';

const renderBioDetails = fellow => (
  <React.Fragment>
    <RenderItem title="Cohort">{fellow.cohort}</RenderItem>
    <RenderItem title="Level">{fellow.level}</RenderItem>
    <RenderItem title="Start Date">
      {fellow.details.actualApprenticeshipStartDate ||
        fellow.details.simulationStartDate}
    </RenderItem>
    <RenderItem title="Estimated End Date">
      {fellow.details.expectedApprenticeshipCompletionDate ||
        fellow.details.expectedSimulationsCompletionDate}
    </RenderItem>
  </React.Fragment>
);

const BioCard = ({ data }) => {
  const { fellow, loading } = data;
  return (
    <div className="bio">
      <div className="my-bio">MY PROFILE</div>
      <div
        className="card-bio"
        id={`fellow-${fellow.email}`}
        role="button"
        tabIndex="0"
      >
        {!loading && fellow.details ? renderBioDetails(fellow) : <Loader />}
      </div>
    </div>
  );
};

BioCard.propTypes = {
  data: PropTypes.shape().isRequired
};

export default BioCard;
