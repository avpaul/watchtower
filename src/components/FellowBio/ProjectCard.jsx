import React from 'react';
import PropTypes from 'prop-types';
import RenderItem from './RenderItem';

const ProjectCard = ({ data, Id }) => (
  <div className="project">
    <div className="my-project">MY PROJECT</div>
    <div className="card-project" id={Id} role="button" tabIndex="0">
      <RenderItem title="Project Name">
        {data.apprenticeshipTeam || data.simulationsTeam}
      </RenderItem>
      <RenderItem title={data.apprenticeshipManager ? 'TTL' : 'LF'}>
        {data.apprenticeshipManager || data.simulationsPM}
      </RenderItem>
      <RenderItem title="PM">--</RenderItem>
    </div>
  </div>
);

ProjectCard.propTypes = {
  Id: PropTypes.string.isRequired,
  data: PropTypes.shape().isRequired
};

export default ProjectCard;
