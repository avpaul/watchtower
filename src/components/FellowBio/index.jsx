import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './FellowBio.css';
import BioCard from './BioCard';
import ProjectCard from './ProjectCard';
/**
 * Class representing the dashboard page
 * @class
 */

export default function FellowBio({ fellow }) {
  return (
    <Fragment>
      <div className="row m-0 fellow-bio-container justify-content-between">
        <BioCard data={fellow} />
        <ProjectCard data={fellow} />
      </div>
    </Fragment>
  );
}

FellowBio.propTypes = {
  fellow: PropTypes.shape().isRequired
};

const mapStateToProps = ({ fellow }) => ({
  fellow
});

export const FellowBioConnected = connect(mapStateToProps)(FellowBio);
