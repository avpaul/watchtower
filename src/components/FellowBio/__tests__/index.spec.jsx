import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import FellowBio, { FellowBioConnected } from '../index';

describe('tests the index component', () => {
  it('renders correctly', () => {
    const props = {
      fellow: {
        loading: true,
        fellow: {}
      },
      getFellowBioAction: jest.fn(),
      user: {
        email: 'roni@mail.com'
      }
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore({ fellow: { fellow: {} } });
    const wrapper = shallow(<FellowBioConnected {...props} store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Fellow Bio is mounted correctly', () => {
    const props = {
      fellow: {
        loading: true,
        fellow: {}
      },
      getFellowBioAction: jest.fn(),
      user: {
        email: 'roni@mail.com'
      }
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore({ fellow: { fellow: {} } });
    const wrapper = shallow(<FellowBioConnected {...props} store={store} />);
    expect(wrapper.find('FellowBio')).toHaveLength(1);
  });
  it('Fellow Bio is mounted correctly', () => {
    const props = {
      fellow: {
        loading: true,
        fellow: {}
      },
      getFellowBioAction: jest.fn(),
      user: {
        email: 'roni@mail.com'
      }
    };
    const wrapper = shallow(<FellowBio {...props} />);
    expect(wrapper.find('BioCard')).toHaveLength(1);
  });
});
