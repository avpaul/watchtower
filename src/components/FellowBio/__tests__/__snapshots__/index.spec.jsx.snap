// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the index component renders correctly 1`] = `
<FellowBio
  dispatch={[Function]}
  fellow={
    Object {
      "fellow": Object {},
    }
  }
  getFellowBioAction={[MockFunction]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "roni@mail.com",
    }
  }
/>
`;
